src/goto-symex/symex_assign.cpp

```diff
@@ -231,10 +232,108 @@ void goto_symext::symex_assign_symbol(
     tmp_ssa_rhs.swap(ssa_rhs);
   }
 
+  // TODO: done via rename
+  // field_sensitivityt::apply(ns, ssa_rhs, false);
+
   state.rename(ssa_rhs, ns);
+
+  ssa_exprt lhs_mod = lhs;
+
+  if(
+    ssa_rhs.id() == ID_byte_update_little_endian ||
+    ssa_rhs.id() == ID_byte_update_big_endian)
+  {
+    byte_update_exprt &bu = to_byte_update_expr(ssa_rhs);
+    exprt be = byte_extract_exprt(
+      bu.id() == ID_byte_update_big_endian ? ID_byte_extract_big_endian
+                                           : ID_byte_extract_little_endian,
+      lhs_mod,
+      bu.offset(),
+      bu.value().type());
+    do_simplify(be);
+
+    if(be.id() == ID_symbol)
+    {
+      ssa_rhs = bu.value();
+      lhs_mod = to_ssa_expr(be);
+    }
+    else if(be.id() == ID_index)
+    {
+      ssa_rhs = bu.value();
+
+      while(be.id() == ID_index)
+      {
+        index_exprt &idx = to_index_expr(be);
+
+#ifdef USE_UPDATE
```

Comment:
If you know what the USE_UPDATE flag means could you document it at the beginning of the file?

Reply:
Done.